File: Data Structures & Algorithms/remove-element/submission-1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        
        // Swaping val to the end 
        
        // int left = 0;
        // int right = nums.size()-1;

        // while(left <= right){

        //     while(nums[left] != val && left <= right){
        //         left++;
        //     }
        //     while(nums[right] == val && left <= right ){
        //         right--;
        //     }
            
        //     if(left <= right ){
        //         int temp = nums[left];
        //         nums[left] = nums[right];
        //         nums[right] = temp;
        //         left++;
        //         right--;
        //     }
        // }
        
        // return right+1;
        
        // better way 
        int left = 0;
        for (int right = 0 ; right < nums.size() ; right++){
            if (nums[right] != val){
                nums[left] = nums[right];
                left++;
            }
        }
        return left;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0 
        right = 0
        total_sum = 0
        min_length = float('inf')
        while right < len(nums) :
            
            total_sum += nums[right]
            
            if total_sum >= target :
                min_length = min(min_length, right - left + 1)
            
            while (total_sum - nums[left]) >= target :
                total_sum -= nums[left]
                left += 1
                min_length = min(min_length, right - left + 1)

            right += 1
        
        if min_length != float('inf') :
            return min_length
        else :
            return 0
        

            
